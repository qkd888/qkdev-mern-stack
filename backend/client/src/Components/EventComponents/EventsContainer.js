import React from "react";
import EventItem from "./EventItem"; // importing the EventItem component

export default function EventsContainer({ events, handleEdit, deleteEvent }) {
  if (events.length > 0) {
    // If the events array has event items then
    // mapping through the array and returning
    // a eventItem component for each item.
    // The EventItem component takes in the eventName,
    // eventDescription and eventDate, handleEdit function and deleteEvent function as props.
    // The event items's id is used as the key for each EventItem component.
    return (
      <div className="event">
        <h1 className="upcomingEvents"> Upcoming Events</h1>
        <div className="eventsContainer">
          {events.map((event) => {
            return (
              <div key={event._id}>
                <EventItem
                  eventName={event.eventName}
                  eventDescription={event.eventDescription}
                  eventDate={event.eventDate}
                  handleEdit={() => handleEdit(event._id)}
                  deleteEvent={() => deleteEvent(event._id)}
                />
              </div>
            );
          })}
        </div>
      </div>
    );
  } else {
    // Else return No upcoming events if the events list is empty
    return <h1>No upcoming events</h1>;
  }
}
