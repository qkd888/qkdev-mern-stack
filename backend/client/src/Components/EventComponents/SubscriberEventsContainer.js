import React from "react";
import SubscriberEvents from "./SubscriberEvents"; // importing the SubscriberEvents component
import trumpet from "./trumpet.jpg"; // importing the trumpet image

export default function SubsciberEventsContainer({ events, name }) {
  if (events.length > 0) {
    // If the events array has event items then
    // mapping through the array and returning
    // a SubscriberEvents component for each item.
    // The SubscriberEvents component takes in  the eventName,
    // eventDescription and eventDate as props.
    //  The event items's id is used as the key for each SubscriberEvent component
    return (
      <div className="event">
        <h1>Welcome, {name}</h1>
        <h2 className="upcomingEventsHeading"> Upcoming Events</h2>
        <div className="subEventsContainer">
          <img
            src={trumpet}
            alt="trumpet"
            className="trumpet"
            width={550}
            height={400}
          />
          <div className="events">
            {events.map((event) => {
              return (
                <div key={event._id}>
                  <SubscriberEvents
                    eventName={event.eventName}
                    eventDescription={event.eventDescription}
                    eventDate={event.eventDate}
                  />
                </div>
              );
            })}
          </div>
        </div>
      </div>
    );
  } else {
    // Else return No upcoming events if the events list is empty
    return <h1>No upcoming events</h1>;
  }
}
