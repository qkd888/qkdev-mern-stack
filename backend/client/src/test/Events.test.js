import React from "react";
import renderer from "react-test-renderer";
import EventsContainer from "../Components/EventComponents/EventsContainer";
import SubscriberEventsContainer from "../Components/EventComponents/SubscriberEventsContainer";

// Testing if Admin Events container renders  properly when passed an empty events list
test("Admin events container renders correctly with empty array", () => {
  const tree = renderer.create(<EventsContainer events={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing if Admin Events container renders  properly when passed a populated events list
test("Admin events container renders correctly with populuated array", () => {
  const events = [
    {
      eventName: "global citizen",
      eventDescription: "Best concert of your life",
      eventDate: "2 December 2018",
      _id: 6,
    },
  ];
  const tree = renderer.create(<EventsContainer events={events} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing if subscriber Events container renders  properly when passed an empty events list

test("Subscriber events renders correctly with empty array", () => {
  const tree = renderer
    .create(<SubscriberEventsContainer events={[]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Testing if subscriber Events container renders  properly when passed a populated events list
test("Subscriber events renders correctly with populuated array", () => {
  const events = [
    {
      eventName: "global citizen",
      eventDescription: "Best concert of your life",
      eventDate: "2 December 2018",
      _id: 6,
    },
  ];
  const tree = renderer.create(<EventsContainer events={events} />).toJSON();
  expect(tree).toMatchSnapshot();
});
