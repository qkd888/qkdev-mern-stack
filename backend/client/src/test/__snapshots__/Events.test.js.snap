// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin events container renders correctly with empty array 1`] = `
<h1>
  No upcoming events
</h1>
`;

exports[`Admin events container renders correctly with populuated array 1`] = `
<div
  className="event"
>
  <h1
    className="upcomingEvents"
  >
     Upcoming Events
  </h1>
  <div
    className="eventsContainer"
  >
    <div>
      <div
        className="eventItem"
      >
        <p>
          global citizen
        </p>
        <p>
          Best concert of your life
        </p>
        <p>
          2 December 2018
        </p>
        <button
          className="edit"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders correctly with empty array 1`] = `
<h1>
  No upcoming events
</h1>
`;

exports[`Renders correctly with populuated array 1`] = `
<div
  className="event"
>
  <h1
    className="upcomingEvents"
  >
     Upcoming Events
  </h1>
  <div
    className="eventsContainer"
  >
    <div>
      <div
        className="eventItem"
      >
        <p>
          global citizen
        </p>
        <p>
          Best concert of your life
        </p>
        <p>
          2 December 2018
        </p>
        <button
          className="edit"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Subscriber events renders correctly with empty array 1`] = `
<h1>
  No upcoming events
</h1>
`;

exports[`Subscriber events renders correctly with populuated array 1`] = `
<div
  className="event"
>
  <h1
    className="upcomingEvents"
  >
     Upcoming Events
  </h1>
  <div
    className="eventsContainer"
  >
    <div>
      <div
        className="eventItem"
      >
        <p>
          global citizen
        </p>
        <p>
          Best concert of your life
        </p>
        <p>
          2 December 2018
        </p>
        <button
          className="edit"
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          className="delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
